File: 856-score-of-parentheses/856-score-of-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var scoreOfParentheses = function (s) {
  let [pwr, ans] = [0, 0];

  for (let i = 1; i < s.length; i++) {
    if (s[i] === "(") pwr++;
    else if (s[i - 1] === "(") ans += 1 << pwr--;
    else pwr--;
  }

  return ans;
};

File: 856-score-of-parentheses/856-score-of-parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        qwr, ans = 0, 0
        for i in range(1, len(s)):
            if s[i] == '(':
                qwr += 1
            elif s[i - 1] == '(':
                ans += 1 << qwr
                qwr -= 1
            else:
                qwr -= 1
        return ans